### PR TITLE
feat: firestore and refactor redux

### PR DESCRIPTION
**Pull request by eldu**
- Refactors store
- Fetches publications from firestore
- Refactor some classes to functions

### CODE REVIEW

**Review by mirestrepo — APPROVED** [review]
Looks great!

File: src/components/ContentPage.js [hunk 1]
```diff
@@ -1,82 +1,63 @@
...
-            </Button>
-          </Link>
-        </div>
-        <div className="d-flex flex-row justify-content-center align-items-center">
-          <div className="pub-title pt-2 bg-primary text-white rounded-circle">
-            <FontAwesomeIcon icon={faBook} />
-          </div>
-          <h1 className="pl-2">Publications</h1>
+import { useDispatch, useSelector } from "react-redux";
+import { fetchPublicationsData } from "../utils/firebase.ts";
```
Comment (tdivoll) [inline src/components/ContentPage.js:11]: Major update, fetching from 🔥 base!

File: src/sagas.js [hunk 2]
```diff
@@ -1,3 +1,7 @@
+/**
+ * Deprecation Notice
+ */
+
```
Comment (tdivoll) [inline src/sagas.js:4]: Should there be another line here with content for this reminder?

File: src/setupTests.js [hunk 3]
```diff
@@ -1,3 +1,7 @@
+/**
+ * Deprecation Notice
+ */
+
```
Comment (tdivoll) [inline src/setupTests.js:4]: Same here, a little more on what will be deprecated?

**Review by tdivoll — CHANGES_REQUESTED** [review]
The code looks good, maybe just add a few lines to the deprecation TODOs to make it easier to remember what they are about?

File: src/sagas.js (hunk 2 above)
Comment (eldu) [reply src/sagas.js:4]: This whole file is going away since redux-saga is no longer recommended!

**Pull request merged**